src/Flex.php

```diff
@@ -863,7 +863,11 @@ public function truncatePackages(PrePoolCreateEvent $event)
         if (!$this->filter) {
             return;
         }
-        $event->setPackages($this->filter->removeLegacyPackages($event->getPackages(), $event->getRequest()->getFixedOrLockedPackages()));
+
+        $rootPackage = $this->composer->getPackage();
+        $lockedPackages = $event->getRequest()->getFixedOrLockedPackages();
```

Comment:
This is not compatible with composer 1, so you may want to make it optional somehow, Semver\Intervals also does not exist in v1.

I guess it's already broken in the current release tho.. Or is this code path only run for composer 2?

Reply:
this codepath runs on c2 only, yes